packages/opencensus-exporter-ocagent/test/test-ocagent.ts

```diff
@@ -517,4 +519,175 @@ describe('OpenCensus Agent Exporter', () => {
       rootSpan.end();
     });
   });
+
+  it('should adapt a span correctly without overflowing trace param limits',
+     (done) => {
+       const rootSpanOptions: TraceOptions = {
+         name: 'root',
+         kind: 'SERVER',
+         spanContext: {
+           traceId: hexId(),
+           spanId: hexId(),
+           traceState: 'foo=bar,baz=buzz',
+           options: 0x1
+         }
+       };
+
+       tracing.tracer.startRootSpan(rootSpanOptions, (rootSpan: RootSpan) => {
+         // Status
+         rootSpan.setStatus(CanonicalCode.OK);
+
+         // Attribute
+         rootSpan.addAttribute('my_first_attribute', 'foo');
+         rootSpan.addAttribute('my_second_attribute', 'foo2');
+
+         // Annotation
+         rootSpan.addAnnotation(
+             'my_annotation',
+             {myString: 'bar', myNumber: 123, myBoolean: true});
+
+         // Metric Event
+         const timeStamp = 123456789;
+         rootSpan.addMessageEvent('MessageEventTypeSent', 'ffff', timeStamp);
```

Comment:
Curious, why is this 'MessageEventTypeSent' and not just 'SENT'?

This would be for a follow up PR, but what would you think about adding enums for span kind, message event type and link type?

Reply:
This is legacy code, I am not sure rational behind this. Created an issue to track this -> https://github.com/census-instrumentation/opencensus-node/issues/333. Thanks 👍